snuba/utils/metrics/__init__.py

```diff
@@ -0,0 +1,59 @@
+from random import random
+from collections import ChainMap
+from typing import Mapping, Optional, Union
+
+
+Tags = Mapping[str, str]
+
+
+from snuba.utils.metrics.backends.abstract import MetricsBackend
+
+
+class Metrics:
+    def __init__(
+        self,
+        backend: MetricsBackend,
+        namespace: Optional[str] = None,
+        tags: Optional[Tags] = None,
+        sample_rates: Optional[Mapping[str, float]] = None,
+    ) -> None:
+        self.__backend = backend
+        self.__namespace = namespace
+        self.__tags = tags
+        self.__sample_rates = sample_rates if sample_rates is not None else {}
+
+    def __apply_namespace(self, name: str) -> str:
+        if self.__namespace is not None:
+            return f"{self.__namespace}.{name}"
+        else:
+            return name
+
+    def __merge_tags(self, tags: Optional[Tags]) -> Tags:
+        return ChainMap(*[d for d in [self.__tags, tags] if d is not None])
+
+    def __is_sampled(self, name: str) -> bool:
+        return self.__sample_rates.get(name, 1.0) >= random()
+
+    def increment(
+        self, name: str, value: Union[int, float] = 1, tags: Optional[Tags] = None
+    ) -> None:
+        if self.__is_sampled(name):
+            self.__backend.increment(
```

Comment:
This function is not covered. This whole class should really be covered with unit tests.